### PR TITLE
Roll up replica sets by deployment

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/HGvCTI96

openshift_web_console

openshift_web_console

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres PTAL

File: app/views/overview/_service.html [hunk 1]
```diff
@@ -10,38 +10,49 @@
...
-  <div ng-repeat="(dcName, deployments) in visibleDeploymentsByConfig" class="deployment-tile-wrapper">
+  <div ng-repeat="(dcName, deployments) in visibleDeploymentsByConfig track by dcName" class="overview-tile-wrapper">
     <!-- visible deployments with a dc -->
     <overview-deployment-config ng-if="dcName"></overview-deployment-config>
     <!-- /visible deployments with a dc -->
 
-    <!-- visible deployments without a dc -->
-    <div class="deployment-tile-wrapper" ng-if="!dcName.length" ng-repeat="deployment in deployments | orderObjectsByDate : true track by (deployment | uid)">
-      <overview-replication-controller></overview-replication-controller>
+    <!-- visible replication controllers without a dc -->
```
Comment (jwforres) [inline app/views/overview/_service.html]: opening and closing comment aren't consistent with naming

File: app/views/overview/_service.html [hunk 2]
```diff
@@ -10,38 +10,49 @@
...
+    <div ng-if="!dcName"
+         ng-repeat="set in deployments | orderObjectsByDate : true track by (deployment | uid)"
```
Comment (jwforres) [inline app/views/overview/_service.html]: you need to fix all the track by references to deployment, they need to be 'set'

File: app/scripts/controllers/deployments.js [hunk 3]
```diff
@@ -91,12 +114,10 @@ angular.module('openshiftConsole')
...
           group: "extensions",
           resource: "replicasets"
-        }, context, function(replicaSets) {
+        }, context, function(replicaSetsData) {
           // TODO: This should be updated to only include replica sets that do not have a deployment.
-          $scope.unfilteredReplicaSets = replicaSets.by("metadata.name");
-          LabelFilter.addLabelSuggestionsFromResources($scope.unfilteredReplicaSets, $scope.labelSuggestions);
-          LabelFilter.setLabelSuggestions($scope.labelSuggestions);
-          $scope.replicaSets = LabelFilter.getLabelSelector().select($scope.unfilteredReplicaSets);
+          replicaSets = replicaSetsData.by("metadata.name");
```
Comment (spadgett) [inline app/scripts/controllers/deployments.js]: Yes, thanks.

File: app/scripts/directives/overview/deployment.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+        $scope.$watch(function() {
+          return _.get($scope, ['deployments', $scope.deploymentName]);
+        }, function() {
+          $scope.deployment = _.get($scope, ['deployments', $scope.deploymentName]);
+          $scope.latestRevision = DeploymentsService.getRevision($scope.deployment);
+        });
+
+        $scope.$watch('scalableReplicaSetsByDeployment', function(replicaSets) {
+          $scope.latestReplciaSet = _.get($scope, ['scalableReplicaSetsByDeployment', $scope.deploymentName]);
```
Comment (spadgett) [inline app/scripts/directives/overview/deployment.js]: At least I typo'd it the same everywhere :/

**Comment by spadgett** [comment]
@jwforres Updated for all comments except for polling. I'm a little concerned about that change. (I know we'll have to do something for IE.)

File: app/scripts/controllers/overview.js [hunk 5]
```diff
@@ -414,12 +477,27 @@ angular.module('openshiftConsole')
...
+      var content = [
+        'monopodsByService',
+        'deploymentConfigsByService',
+        'deploymentsByService',
+        'replicationControllersByService',
+        'replicaSetsByService',
+        'petSetsByService'
+      ];
+      return _.some(content, function(contentByService) {
+        var unservicedContent = _.get($scope, [contentByService, ''], []);
```
Comment (spadgett) [inline app/scripts/controllers/overview.js]: Fixed

File: app/scripts/controllers/overview.js [hunk 6]
```diff
@@ -462,25 +540,25 @@ angular.module('openshiftConsole')
           groupServices();
           groupPods();
           groupDeploymentConfigs();
-          groupDeployments();
+          groupReplicationControllers();
```
Comment (spadgett) [inline app/scripts/controllers/overview.js]: Fixed

File: app/views/overview/_service.html (hunk 1 above)
Comment (spadgett) [inline app/views/overview/_service.html]: Updated the var names

File: app/views/overview/_service.html (hunk 2 above)
Comment (spadgett) [inline app/views/overview/_service.html]: Fixed

**Comment by spadgett** [comment]
@jwforres updated polling

File: app/scripts/controllers/overview.js [hunk 7]
```diff
@@ -540,8 +539,9 @@ angular.module('openshiftConsole')
           groupServices();
           groupPods();
           groupDeploymentConfigs();
-          groupReplicationControllers();
-          groupReplicaSets();
+          groupReplicationControllersByService();
+          groupReplicationControllersByDC();
```
Comment (jwforres) [inline app/scripts/controllers/overview.js]: does this really need to be called here? groupRCByDC doesn't depend on services that i can see

File: app/views/overview/_service.html [hunk 8]
```diff
@@ -1,4 +1,4 @@
-<div ng-if="!(visibleDeploymentsByConfig | hashSize) && !(visibleReplicaSets | hashSize) && !(petSetsByService[service.metadata.name] | hashSize) && !(monopodsByService[service.metadata.name] | hashSize)" class="no-deployments-block">
+<div ng-if="!(visibleReplicaionControllersByDC | hashSize) && !(visibleReplicaSets | hashSize) && !(petSetsByService[service.metadata.name] | hashSize) && !(monopodsByService[service.metadata.name] | hashSize)" class="no-deployments-block">
```
Comment (jwforres) [inline app/views/overview/_service.html]: typo visibleReplicationControllersByDC its in a couple places

File: app/scripts/controllers/overview.js (hunk 7 above)
Comment (spadgett) [inline app/scripts/controllers/overview.js]: https://github.com/openshift/origin-web-console/pull/617/files#diff-7878fa660740a71bf5dbfa99b53c8b43R139

File: app/views/overview/_service.html (hunk 8 above)
Comment (spadgett) [inline app/views/overview/_service.html]: Will fix

**Comment by spadgett** [comment]
Fixed the typo. (I'll squash before merge.)

**Comment by spadgett** [comment]
Thanks @jwforres 

[merge]

**Pull request merged**